numpy/core/src/multiarray/buffer.c

```diff
@@ -1057,7 +1057,7 @@ _descriptor_from_pep3118_format_fast(char const *s, PyObject **result)
     else {
         *result = (PyObject*)PyArray_DescrNewByteorder(descr, byte_order);
         Py_DECREF(descr);
-        if (result == NULL) {
+        if (*result == NULL) {
```

Comment:
Well, clang is smart enough to notice that it was always false, to bad GCC running in CI is not :)